src/api/internal.rs

```diff
@@ -587,21 +608,23 @@ impl<T: Pixel> ContextInner<T> {
 
     #[cfg(feature = "dump_lookahead_data")]
     {
+      let data_location =
+        Self::build_dump_properties().as_path().display().to_string();
```

Comment:
It would be more correct to avoid path-to-regular-string conversion and instead use `Path::join()`. `ImageBuffer::save()` does accept an `AsRef<Path>` rather than a `&str`. This way there will be no issues with invalid-UTF8 paths.